core/src/main/kotlin/model/additionalExtras.kt

```diff
@@ -123,4 +123,17 @@ data class ConstructorValues(val values: SourceSetDependent<List<Expression>>) :
     }
 
     override val key: ExtraProperty.Key<DEnumEntry, ConstructorValues> = ConstructorValues
-}
\ No newline at end of file
+}
+
+data class AncestryNode(
+    val typeConstructor: TypeConstructor,
+    val superclass: AncestryNode?,
+    val interfaces: List<AncestryNode>,
+) {
+    fun allImplementedInterfaces(): List<TypeConstructor> {
+        fun traverseInterfaces(ancestry: AncestryNode): List<TypeConstructor> =
+            ancestry.interfaces.flatMap { listOf(it.typeConstructor) + traverseInterfaces(it) } +
+                (ancestry.superclass?.let(::traverseInterfaces) ?: emptyList())
+        return traverseInterfaces(this).distinct()
+    }
+}
```

Comment:
Should this be in `additionalExtras.kt` if it's not an extra?

Reply:
Hard to tell, I can move it to another file, though I felt like the file is for containing addition model elements (I know that all are extra related). Should I move it to some new file like `AncestryNode.kt`?